test/apps/iodemo/io_demo.cc

```diff
@@ -566,6 +578,20 @@ class P2pDemoCommon : public UcxContext {
             _pool.put(this);
         }
 
+        inline void set_buffer(void *buffer)
+        {
+            assert(_zcopy);
+
+            _buffer = buffer;
+        }
+
+        inline void set_capacity(size_t capacity)
+        {
+            assert(_zcopy);
+
+            _capacity = capacity;
+        }
```

Comment:
capacity should be hidden from a user and updated automatically by initialization or resize